pkg/controllers/work/patch_util_test.go

```diff
@@ -0,0 +1,44 @@
+package controllers
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestSetModifiedConfigurationAnnotation(t *testing.T) {
+	smallObj, _, _, err := createObjAndDynamicClient(testManifest.Raw)
+	if err != nil {
+		t.Errorf("failed to create obj and dynamic client: %s", err)
+	}
+	largeObj, err := createLargeObj()
+	if err != nil {
+		t.Errorf("failed to create large obj: %s", err)
+	}
+
+	tests := map[string]struct {
+		obj      runtime.Object
+		wantBool bool
+		wantErr  error
+	}{
+		"last applied config annotation is set": {
+			obj:      smallObj,
+			wantBool: true,
+			wantErr:  nil,
+		},
+		"last applied config annotation is set to an empty string": {
+			obj:      largeObj,
+			wantBool: false,
+			wantErr:  nil,
+		},
+	}
+
+	for testName, testCase := range tests {
+		t.Run(testName, func(t *testing.T) {
+			gotBool, gotErr := setModifiedConfigurationAnnotation(testCase.obj)
+			assert.Equalf(t, testCase.wantBool, gotBool, "got bool not matching for Testcase %s", testName)
```

Comment:
Can we try to avoid the Assert libraries in the UT?

"Avoid the use of 'assert' libraries to help your tests. "
https://github.com/golang/go/wiki/TestComments#assert-libraries